src/editor.rs

```diff
@@ -267,32 +267,21 @@ fn parse_command(keymap: &KeyMap, command: &str) -> Option<Action> {
     let mut command_char_iter = command.chars();
 
     // Consume the first char of the command
-    let c = command_char_iter.next()?;
-
-    match keymap.get(&c) {
-        // "q" quits Sapling
-        Some(Command::Quit) => Some(Action::Quit),
-        Some(Command::InsertChild) => {
-            // Consume the second char of the iterator
-            command_char_iter.next().map(Action::InsertChild)
+    return if let Some(command) = command_char_iter.next().and_then(|c| keymap.get(&c)) {
+        match command {
+            // "q" quits Sapling
+            Command::Quit => Some(Action::Quit),
+            Command::InsertChild => command_char_iter.next().map(Action::InsertChild),
+            Command::InsertBefore => command_char_iter.next().map(Action::InsertBefore),
+            Command::InsertAfter => command_char_iter.next().map(Action::InsertAfter),
+            Command::Replace => command_char_iter.next().map(Action::Replace),
+            Command::MoveCursor(direction) => Some(Action::MoveCursor(*direction)),
+            Command::Undo => Some(Action::Undo),
+            Command::Redo => Some(Action::Redo),
         }
-        Some(Command::InsertBefore) => {
-            // Consume the second char of the iterator
```

Comment:
I like how compact the code now is, but I think we should move it somewhere else rather than simply removing it.  Perhaps put a comment explaining the purpose of `command_char_iter.next().map(...)` above the `match command {` line.